### PR TITLE
Support for cmat3 d block

### PR DESCRIPTION
**Pull request by diptighosalkar**
Fixed define(), get() and set() of cmat3D block.
Added support for Colormap values in block

### CODE REVIEW

File: webapp/data_structures/CMAT3D.js [hunk 1]
```diff
@@ -1,72 +1,99 @@
...
-    return new BasicBlock(this.x);
+    CMAT3D.prototype.define = function CMAT3D() {
+        this.cmin = 0;
+        this.cmax = 100;
+        this.colormap_string = "jetcolormap(25)";
+        var colormap = get_colormap(this.colormap_string);
+        var size_c = 25;
+        this.vec_x = -1;
+        this.vec_y = -1;
+        this.size_x = 1;
```
Comment (sunilshetye) [inline webapp/data_structures/CMAT3D.js]: use var instead of this here

File: webapp/data_structures/CMAT3D.js [hunk 2]
```diff
@@ -1,72 +1,99 @@
...
+        this.size_y = 1;
```
Comment (sunilshetye) [inline webapp/data_structures/CMAT3D.js]: use var instead of this here

File: webapp/data_structures/CMAT3D.js [hunk 3]
```diff
@@ -1,72 +1,99 @@
...
+        var model = scicos_model();
+        model.sim = list(new ScilabString(["cmat3d"]), new ScilabDouble([4]));
+        model.in = new ScilabDouble([-1]);
+        model.in2 = new ScilabDouble([-2]);
+        model.intyp = new ScilabDouble([1]);
+        model.evtin = new ScilabDouble([1]);
+        model.ipar = new ScilabDouble([this.cmin], [this.cmax], [size_c], [this.size_x], [this.size_y]);
```
Comment (sunilshetye) [inline webapp/data_structures/CMAT3D.js]: size_x and size_y here

File: webapp/data_structures/CMAT3D.js [hunk 4]
```diff
@@ -1,72 +1,99 @@
...
+        }
+        var cmin_1 = parseFloat(arguments[0]["cmin"]);
+	    var cmax_1 = parseFloat(arguments[0]["cmax"]);
+	    if(cmax_1 <= cmin_1){
+            alert("Error with minimum and maximum value");
+            throw "incorrect";
+        }
+        this.cmin = cmin_1;
+        this.cmax = cmax_1;
+        this.size_x = size(this.vec_x,"*");
```
Comment (sunilshetye) [inline webapp/data_structures/CMAT3D.js]: use var instead of this here

File: webapp/data_structures/CMAT3D.js [hunk 5]
```diff
@@ -1,72 +1,99 @@
...
+        var ipar = new ScilabDouble([this.cmin],[this.cmax],[size_c],[this.size_x]);
```
Comment (sunilshetye) [inline webapp/data_structures/CMAT3D.js]: size_x here

also, size_y is missing

Comment (diptighosalkar) [reply webapp/data_structures/CMAT3D.js]: In cmat3d.sci file, inside set function's ipar object "size_y" is not there , so actually it is not missing but not there only.

Comment (sunilshetye) [reply webapp/data_structures/CMAT3D.js]: okay

**Pull request merged**